### PR TITLE
Feature/pin 10851 reviewers edge cases

### PR DESCRIPTION
**Pull request by l-giacomini**
## 🔗 Issue
[PIN-10851](https://pagopa.atlassian.net/browse/PIN-10851)

## 📝 Description / Context

This PR improves the reviewer assignment step in the Risk Analysis flow by handling reviewers who are no longer available in the institution. It updates the validation behavior to distinguish between fully and partially removed reviewer assignments, providing immediate feedback and a warning when applicable.

## 🛠 List of changes

* Updated the reviewer autocomplete to display only currently available reviewers.
* Added validation triggered on render when all previously assigned reviewers are no longer available.
* Updated reviewer validation to distinguish between:
  * All previously assigned reviewers being unavailable, requiring the selection of new reviewers.
  * Some previously assigned reviewers being unavailable while others remain available, allowing the user to proceed with a warning.
* Added a warning alert for partially removed reviewer assignments, with the option to dismiss it.
* Updated the info alert UI for cases where no reviewers are available from Selfcare, preserving the existing behavior.
* Added and updated unit tests to cover removed reviewers, immediate validation, partial removal warnings, and the existing no-reviewer scenarios.

## 🧪 How to test

* Navigate to the Risk Analysis assignment step for a purpose with an existing reviewer assignment.
* Verify that the autocomplete only offers currently available reviewers.
* Verify that when all previously assigned reviewers have been removed but other reviewers are available, the validation error is displayed immediately.
* Verify that when only some previously assigned reviewers have been removed and at least one remains available, the warning alert is displayed without blocking the form.
* Verify that the warning can be dismissed.
* Verify that submitting with an invalid removed reviewer assignment is blocked.
* Verify that when no reviewers are available from Selfcare, the existing info alert is displayed and the form is hidden.

[PIN-10851]: https://pagopa.atlassian.net/browse/PIN-10851?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/pages/ConsumerPurposeEditPage/components/PurposeEditStepAssignment/PurposeEditStepAssignmentForm.tsx [hunk 1]
```diff
@@ -279,6 +303,9 @@ const PurposeEditStepAssignmentForm: React.FC<PurposeEditStepAssignmentFormProps
                   <>
                     <Typography variant="body2" fontWeight={600}>
```
Comment (borgesis95) [inline src/pages/ConsumerPurposeEditPage/components/PurposeEditStepAssignment/PurposeEditStepAssignmentForm.tsx:295]: MUI's default variant mapping renders body2 as <p> so this nests a <p> inside another <p> and it break a11y 

```suggestion
                     <Typography variant="body2" fontWeight={600}>
                      {t(`reviewerField.label.${reviewerLabelKey}`)}
                    </Typography>
                    <Typography variant="body2" fontWeight={400} sx={{ mt: 1 }}>
                      {t('reviewerField.infoLabel')}
                    </Typography>
```

File: src/pages/ConsumerPurposeEditPage/components/PurposeEditStepAssignment/PurposeEditStepAssignmentForm.tsx [hunk 2]
```diff
@@ -1,9 +1,10 @@
-import React from 'react'
+import React, { useState } from 'react'
```
Comment (borgesis95) [inline src/pages/ConsumerPurposeEditPage/components/PurposeEditStepAssignment/PurposeEditStepAssignmentForm.tsx:1]: Would it make sense to extract the reviewer-availability logic (`removedReviewers`, `removedReviewerIds`, `hasRemovedReviewers`, `hasAvailableAssignedReviewers`, `hasLostItsOnlyReviewers`etc...)  into a dedicated hook, e.g. useReviewerAssignmentState(purpose, reviewers)? It's a fair amount of derived state and validation logic living inline in the form component, and pulling it out would make PurposeEditStepAssignmentForm easier to read and let us unit-test the removed/partial-reviewer edge cases directly against the hook instead of only through the rendered form.

File: src/pages/ConsumerPurposeEditPage/components/PurposeEditStepAssignment/PurposeEditStepAssignmentForm.tsx (hunk 1 above)
Comment (l-giacomini) [reply src/pages/ConsumerPurposeEditPage/components/PurposeEditStepAssignment/PurposeEditStepAssignmentForm.tsx:295]: Fixed in: https://github.com/pagopa/pdnd-interop-frontend/pull/2225/changes/98ddf01ebd30a99911001b06938c51349979be20

File: src/pages/ConsumerPurposeEditPage/components/PurposeEditStepAssignment/PurposeEditStepAssignmentForm.tsx [hunk 3]
```diff
@@ -290,7 +317,28 @@ const PurposeEditStepAssignmentForm: React.FC<PurposeEditStepAssignmentFormProps
                         validate: validateReviewerIds,
                       }}
                     />
-                  </>
+
+                    {hasRemovedReviewers &&
+                      hasAvailableAssignedReviewers &&
+                      showPartiallyRemovedReviewersAlert && (
+                        <Alert
```
Comment (borgesis95) [inline src/pages/ConsumerPurposeEditPage/components/PurposeEditStepAssignment/PurposeEditStepAssignmentForm.tsx:315]: Can I ask you to start to use this? (don't hate me) 

```suggestion
                         <MIAlert
                        variant="default"
                        severity="warning"
                        title={t('reviewerField.partiallyRemovedWarningTitle')}
                        description={t('reviewerField.partiallyRemovedWarningLabel')}
                        action={{
                          label: t('reviewerField.dismissButtonLabel'),
                          onClick: () => setShowPartiallyRemovedReviewersAlert(false),
                        }}
                      />
```

File: src/pages/ConsumerPurposeEditPage/components/PurposeEditStepAssignment/PurposeEditStepAssignmentForm.tsx (hunk 2 above)
Comment (l-giacomini) [reply src/pages/ConsumerPurposeEditPage/components/PurposeEditStepAssignment/PurposeEditStepAssignmentForm.tsx:1]: Added custom hook here: https://github.com/pagopa/pdnd-interop-frontend/pull/2225/changes/156da58f706228d770c941546d179af18afdc91f

File: src/pages/ConsumerPurposeEditPage/components/PurposeEditStepAssignment/PurposeEditStepAssignmentForm.tsx (hunk 3 above)
Comment (l-giacomini) [reply src/pages/ConsumerPurposeEditPage/components/PurposeEditStepAssignment/PurposeEditStepAssignmentForm.tsx]: Fixed in: https://github.com/pagopa/pdnd-interop-frontend/pull/2225/changes/b2451d996c2495fff188eb102a3ba50081356a22
